src/Internal/Contract/QueryBackend.purs

```diff
@@ -0,0 +1,156 @@
+module Ctl.Internal.Contract.QueryBackend
+  ( BlockfrostBackend
+  , CtlBackend
+  , QueryBackend(BlockfrostBackend, CtlBackend)
+  , QueryBackendLabel(BlockfrostBackendLabel, CtlBackendLabel)
+  , QueryBackendParams(BlockfrostBackendParams, CtlBackendParams)
+  , QueryBackends
+  , class HasQueryBackendLabel
+  , backendLabel
+  , defaultBackend
+  , lookupBackend
+  , mkBackendParams
+  , mkSingletonBackendParams
+  , mkCtlBackendParams
+  , mkBlockfrostBackendParams
+  ) where
+
+import Prelude
+
+import Ctl.Internal.QueryM (DatumCacheWebSocket, OgmiosWebSocket)
+import Ctl.Internal.QueryM.ServerConfig (ServerConfig)
+import Data.Array (nub) as Array
+import Data.Array ((:))
+import Data.Foldable (class Foldable, foldMap, foldl, foldr, length)
+import Data.Map (Map)
+import Data.Map (empty, insert, lookup) as Map
+import Data.Maybe (Maybe(Just))
+import Data.Traversable (class Traversable, sequence, traverse)
+import Effect (Effect)
+import Effect.Exception (throw)
+
+--------------------------------------------------------------------------------
+-- QueryBackends
+--------------------------------------------------------------------------------
+
+-- | A generic type to represent a choice of backend with a set of fallback
+-- | backends when an operation is not supported by the default.
+data QueryBackends (backend :: Type) =
+  QueryBackends backend (Map QueryBackendLabel backend)
+
+-- Functor breaks this datatype...
```

Comment:
What does this mean?

Reply:
Addressed by https://github.com/Plutonomicon/cardano-transaction-lib/pull/1314